prices/getawsprices.go

```diff
@@ -53,5 +37,99 @@ func GetPricesMapFromFile(filename string) (PricesMap, error) {
 	if err != nil {
 		return nil, err
 	}
-	return getPricesMap(bytes.NewReader(bs))
+	return populatePricesMap(bytes.NewReader(bs))
+}
+
+func DownloadPricesMap(url string) (PricesMap, error) {
+	if url == "" {
+		return PricesMap{}, nil
+	}
+
+	// TODO: working TLS
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{Transport: tr}
+
+	res, err := client.Get(url)
+	if err != nil {
+		log.Error(err.Error())
+	}
+	defer res.Body.Close()
+	return populatePricesMap(res.Body)
+}
+
+func populatePricesMap(r io.Reader) (PricesMap, error) {
```

Comment:
Why not decode in to a struct with the regular json package? It looks like the price data is in a common format.

```
type ProductPriceData struct {
	Attributes struct {
		ClockSpeed            string `json:"clockSpeed"`
		CurrentGeneration     string `json:"currentGeneration"`
		InstanceFamily        string `json:"instanceFamily"`
		InstanceType          string `json:"instanceType"`
		LicenseModel          string `json:"licenseModel"`
		Location              string `json:"location"`
		LocationType          string `json:"locationType"`
		Memory                string `json:"memory"`
		NetworkPerformance    string `json:"networkPerformance"`
		OperatingSystem       string `json:"operatingSystem"`
		Operation             string `json:"operation"`
		PhysicalProcessor     string `json:"physicalProcessor"`
		PreInstalledSw        string `json:"preInstalledSw"`
		ProcessorArchitecture string `json:"processorArchitecture"`
		Servicecode           string `json:"servicecode"`
		Storage               string `json:"storage"`
		Tenancy               string `json:"tenancy"`
		Usagetype             string `json:"usagetype"`
		Vcpu                  string `json:"vcpu"`
	} `json:"attributes"`
	ProductFamily string `json:"productFamily"`
	Sku           string `json:"sku"`
}
type PriceData struct {
	Disclaimer    string `json:"disclaimer"`
	FormatVersion string `json:"formatVersion"`
	OfferCode     string `json:"offerCode"`
	Products      map[string]ProductPriceData `json:"products"`
	PublicationDate string `json:"publicationDate"`
	Version         string `json:"version"`
}

```